### PR TITLE
Limit shared requests for AdSense reports to metrics and dimensions used by the Site Kit dashboard.

### PR DESCRIPTION
**Pull request by techanvil**
## Summary

Addresses issue:

- #5712 

## Relevant technical choices

- The IB has been diverged from in some areas to align with the review changes made to the related PR https://github.com/google/site-kit-wp/pull/5721.

## PR Author Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [ ] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

File: includes/Modules/AdSense.php [hunk 1]
```diff
@@ -838,4 +858,84 @@ public function check_service_entity_access() {
...
+	 */
+	protected function validate_report_metrics( $metrics ) {
+		if ( false === $this->is_using_shared_credentials ) {
+			return null;
+		}
+
+		$valid_metrics = apply_filters(
+			'googlesitekit_shareable_adsense_metrics',
+			array(
+				'ESTIMATED_EARNINGS',
```
Comment (techanvil) [inline includes/Modules/AdSense.php:878]: As per https://github.com/google/site-kit-wp/issues/5712#issuecomment-1222341102, please take the time to double-check the codebase and verify this list of metrics.

File: includes/Modules/AdSense.php [hunk 2]
```diff
@@ -838,4 +858,84 @@ public function check_service_entity_access() {
...
+	 */
+	protected function validate_report_dimensions( $dimensions ) {
+		if ( false === $this->is_using_shared_credentials ) {
+			return null;
+		}
+
+		$valid_dimensions = apply_filters(
+			'googlesitekit_shareable_adsense_dimensions',
+			array(
+				'DATE',
```
Comment (techanvil) [inline includes/Modules/AdSense.php:920]: As per https://github.com/google/site-kit-wp/issues/5712#issuecomment-1222341102, please take the time to double-check the codebase and verify this dimension is the only one in current use.

**Comment by aaemnnosttv** [comment]
@techanvil this should be ready to finish up. Please see the final changes in #5727.

**Review by eugene-manuilov — APPROVED** [review]
LGTM

File: includes/Modules/AdSense.php [hunk 3]
```diff
@@ -276,11 +278,29 @@ protected function create_data_request( Data_Request $data ) {
 
 				$metrics = $this->parse_string_list( $data['metrics'] );
 				if ( ! empty( $metrics ) ) {
+					try {
+						$this->validate_report_metrics( $metrics );
+					} catch ( Invalid_Report_Metrics_Exception $exception ) {
+						return new WP_Error(
+							'invalid_analytics_report_metrics',
```
Comment (aaemnnosttv) [inline includes/Modules/AdSense.php:285]:
```suggestion
							'invalid_adsense_report_metrics',
```

File: includes/Modules/AdSense.php [hunk 4]
```diff
@@ -276,11 +278,29 @@ protected function create_data_request( Data_Request $data ) {
...
 					$args['metrics'] = $metrics;
 				}
 
 				$dimensions = $this->parse_string_list( $data['dimensions'] );
 				if ( ! empty( $dimensions ) ) {
+					try {
+						$this->validate_report_dimensions( $dimensions );
+					} catch ( Invalid_Report_Dimensions_Exception $exception ) {
+						return new WP_Error(
+							'invalid_analytics_report_dimensions',
```
Comment (aaemnnosttv) [inline includes/Modules/AdSense.php:299]:
```suggestion
							'invalid_adsense_report_dimensions',
```

File: includes/Modules/Analytics.php [hunk 5]
```diff
@@ -1361,7 +1361,7 @@ public function check_service_entity_access() {
 	 */
 	protected function validate_report_metrics( $metrics ) {
 		if ( false === $this->is_using_shared_credentials ) {
-			return null;
```
Comment (aaemnnosttv) [inline includes/Modules/Analytics.php:1364]: 👍

**Review by aaemnnosttv — CHANGES_REQUESTED** [review]
LGTM @techanvil, just a few minor details that seem to have been missed.

**Comment by techanvil** [comment]
> LGTM @techanvil, just a few minor details that seem to have been missed.

D'oh, thanks for the spot. Have applied the fixes.

**Review by aaemnnosttv — APPROVED** [review]
LGTM, thanks @techanvil !

**Comment by aaemnnosttv** [comment]
I changed the base back to develop as the release is already set to go and this has been downgraded in priority as well.

**Pull request merged**